services/QuillLMS/app/controllers/partner_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class PartnerController < ApplicationController
+  include ForceDbWriterRole
+
+  around_action :force_writer_db_role, only: [:activity_session]
+  before_action :confirm_valid_partner, only: [:activity_session]
+  before_action :validate_required_params, only: [:activity_session]
+
+  def activity_session
```

Comment:
- routes.rb declares this to be a GET, but it should be a POST, since it's performing a write.
- it's semantically better for a POST controller method to include a verb, since noun method names indicate a GET. Thus:  `create_partner_activity_session`